### PR TITLE
101-dokku - Deploy a Dokku Instance

### PR DESCRIPTION
**Pull request by sedouard**
Easily deploy a [Dokku](http://progrium.viewdocs.io/dokku/) instance on a VM. Git-push any heroku comaptible app (including ruby, nodejs, go, php, python and anything else).

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@sedouard__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by sedouard** [comment]
I'm not sure why @azurecia thinks I need a CLA. I'm publicly in the Microsoft org so it should pickup I don't need one.

File: 101-dokku/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        "Standard_D2",
+        "Standard_D3",
+        "Standard_D4"
+      ],
+      "metadata": {
+        "description": "Size of vm"
+      }
+    },
+    "location": {
+        "type": "String",
```
Comment (singhkay) [inline 101-dokku/azuredeploy.json]: lower case "string"

File: 101-dokku/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "api-version": "2015-06-15",
+    "sshKeyPath": "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+    "githubPath": "https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/101-dokku/"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline 101-dokku/azuredeploy.json]: You have a variable api-version but it's not being used

**Comment by singhkay** [comment]
@sedouard Let's rename this template from 101 since those are reserved for platform features to something along the lines of [application-scenario]. Maybe dokku-vm ?

File: 101-dokku/azuredeploy.json (hunk 2 above)
Comment (sedouard) [inline 101-dokku/azuredeploy.json]: It's being used on line 229:
```json
{
      "type": "Microsoft.Compute/virtualMachines/extensions",
      "name": "[concat(variables('vmName'),'/initdokku')]",
      "apiVersion": "[variables('api-version')]",
      "location": "[parameters('location')]",
      "dependsOn": [
        "[concat('Microsoft.Compute/virtualMachines/', variables('vmName'))]"
      ],
      "properties": {
        "publisher": "Microsoft.OSTCExtensions",
        "type": "CustomScriptForLinux",
        "typeHandlerVersion": "1.2",
        "settings": {
          "fileUris": [
            "[concat(variables('githubPath'),'deploy_dokku.sh')]"
          ],
          "commandToExecute": "[concat('sh deploy_dokku.sh')]"
        }
      }
    }
```

Comment (sedouard) [inline 101-dokku/azuredeploy.json]: Though I'll change all `apiVersion` fields to use the variable.

**Comment by sedouard** [comment]
Gonna close this PR and submit a new one from a branch thats called `dokku_vm`

**Pull request closed without merging**